### ISSUE TITLE
SkipFlagParsing doesn't work if first argument is a flag

### ISSUE THREAD

**Issue by muja** [codegangsta/cli#208]
It prints 'Incorrect usage' if the first argument is a flag:

```go
app := cli.NewApp()
app.Commands = []cli.Command{
  {
    Name: "test",
    Usage: "cli testing",
    Action: func(c *cli.Context) {
      for _, x := range c.Args() {
        fmt.Fprintln(os.Stdout, x)
      }
    },
    SkipFlagParsing: true,
  },
}
app.Run(os.Args)
```

```
$ go run main.go test arg --flag
arg
--flag

$ go run main.go test --flag arg
Incorrect Usage.

NAME:
   test - cli testing

USAGE:
   command test [arguments...]
```

**Comment by jszwedko** [comment]
Looking at the implementation of `SkipFlagParsing` and when it was introduced, I think that having this field at all is confusing and keeping the default behaviour of requiring a `--` before flags that start with a `-` is better and consistent with CLI interfaces in general. See #209.

**Comment by nathanleclaire** [comment]
+1 to this issue, it is proving problematic to our use case as well.  I disagree that the `--` operator should be required -- in some situations simply disabling flag parsing results in a nicer experience for the end user, e.g. `docker-machine ssh default df -h`.

**Comment by nathanleclaire** [comment]
I am looking into trying to debug the issue myself but the logic is proving to be a little hard to follow in the relevant bits in `command.go`.

**Comment by jszwedko** [comment]
Closed by #283
